src/sudoku.rs

```diff
@@ -136,13 +136,24 @@ impl<const N: usize> Board<N> {
             for value in &row.0 {
                 scorer.check(*value);
             }
+
             total_duplicates += scorer.score();
         }
 
         total_duplicates
     }
 
-    fn count_box_duplicates(&self) -> u8 {
+    /// Counts box duplicates.
+    ///
+    /// Iterates through the Board's sub-boxes to count duplicated values.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the size of the Board, N, is not a perfect square >= 4 or <=
+    /// 25.
+    #[inline]
+    #[must_use]
+    pub fn count_box_duplicates(&self) -> u8 {
```

Comment:
I believe this needs to be public in order to benchmark. I'd be curious to know if it doesn't need to be public so we can hide more of these implementation details.

Reply:
Also, I'm not particularly keen to switch to nightly because I'm hoping the lessons I learn will translate to usable knowledge at work where stability is very important. I think I read somewhere that benchmarking in nightly looks pretty good.